packages/medusa/src/strategies/batch-jobs/product/import.ts

```diff
@@ -405,6 +411,16 @@ class ProductImportStrategy extends AbstractBatchJobStrategy {
           )
         }
 
+        if (
+          productOp["product.collection.handle"] != null &&
+          productOp["product.collection.handle"] !== ""
+        ) {
+          productData["collection"] =
+            await productCollectionServiceTx.retrieveByHandle(
+              productOp["product.collection.handle"] as string
+            )
```

Comment:
nit: `collection` is not a property of `CreateProductInput`, we have used `collection_id` until now. We should either allow `Collection` to be passed on "dot" the id from the result of `retrieveByHandle`.